modin/experimental/xgboost/xgboost.py

```diff
@@ -0,0 +1,499 @@
+# Licensed to Modin Development Team under one or more contributor license agreements.
+# See the NOTICE file distributed with this work for additional information regarding
+# copyright ownership.  The Modin Development Team licenses this file to you under the
+# Apache License, Version 2.0 (the "License"); you may not use this file except in
+# compliance with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+
+import os
+import time
+import logging
+from threading import Thread
+from typing import Dict, Optional
+from multiprocessing import cpu_count
+
+import xgboost as xgb
+import numpy as np
+import pandas
+
+from modin.developer import unwrap_row_partitions
+
+if os.environ.get("MODIN_ENGINE", "Ray").title() == "Ray":
+    import ray
+    from ray.services import get_node_ip_address
+else:
+    raise ValueError("Current version supports only Ray engine as MODIN_ENGINE.")
+
+LOGGER = logging.getLogger("[xgboost.modin]")
+
+
+def _start_rabit_tracker(num_workers: int):
+    """Start Rabit tracker. The workers connect to this tracker to share
+    their results."""
+    host = get_node_ip_address()
+
+    env = {"DMLC_NUM_WORKER": num_workers}
+    rabit_tracker = xgb.RabitTracker(hostIP=host, nslave=num_workers)
+
+    # Get tracker Host + IP
+    env.update(rabit_tracker.slave_envs())
+    rabit_tracker.start(num_workers)
+
+    # Wait until context completion
+    thread = Thread(target=rabit_tracker.join)
+    thread.daemon = True
+    thread.start()
```

Comment:
This is need for correct lifetime of `rabit_tracker`.

Reply:
It would achieve almost the same effect by just adding it to a global variable.
Also, given `RabitTracker` uses a thread internally which target has its contained as a closure variable these 3 lines seem pointless to me.

Reply:
Function `stop_rabit_tracker` was added for stopping tracker after usage.